cli/src/command.rs

```diff
@@ -215,6 +215,68 @@ fn ensure_dev(spec: &Box<dyn service::ChainSpec>) -> std::result::Result<(), Str
 	}
 }
 
+/// Runs a performance check via compiling sample wasm code with a timeout.
+/// Only available in release build since the check would take too much time otherwise.
+/// Returns `Ok` if the check has been passed previously.
+#[cfg(not(debug_assertions))]
+fn host_perf_check() -> Result<()> {
+	use polkadot_node_core_pvf::sp_maybe_compressed_blob;
+	use std::{fs::OpenOptions, path::Path, time::Duration};
+
+	const PERF_CHECK_TIME_LIMIT: Duration = Duration::from_secs(20);
+	const CODE_SIZE_LIMIT: usize = 1024usize.pow(3);
+	const WASM_CODE: &[u8] = include_bytes!(
+		"../../target/release/wbuild/kusama-runtime/kusama_runtime.compact.compressed.wasm"
+	);
+	const CHECK_PASSED_FILE_NAME: &str = ".perf_check_passed";
+
+	// We will try to save a dummy file to the same path as the polkadot binary
+	// to make it independent from the current directory.
+	let check_passed_path = std::env::current_exe()
```

Comment:
Adding the `.`-file in the same directory seems very ugly to me, `/var/run/polkadot` or similar seem more rational to me. We are already storing plenty of data there. The file should also include some way to verify if it was copied from a different machine or not and hence invalidating it.

Reply:
Do you think it's worth trying to verify that the dummy file is not "forged"? You can always modify sources and drop the benchmark if you want.
My point here is that one way or another this check will be optional, especially since one would want to turn it off completely for development purposes.

Reply:
My point being about accidentally copying the i-already-ran-this file from one machine to another. There should be a safeguard that detects this and warns the user (not so much error, just warn) about that fact.

The point is not to prevent launch without having it - in fact there should be an explicit flag to disable this behavior - it's about fool-proofing.

Reply:
For validating we could go with a simple hostname, if it's not sufficient there's also unix machine id,
What about `/var/run/polkadot`? Are users supposed to create it themselves?

Reply:
Will this one do? 9414d2fbf3ce5c2e1db7a3f6bdfd4e1e0bce14a3